proto/insonmnia.go

```diff
@@ -0,0 +1,152 @@
+package sonm
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	ds "github.com/c2h5oh/datasize"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+var snmPriceRe = regexp.MustCompile(`(\S+)(\s*)(snm)(\S+)`)
```

Comment:
I strictly agains putting regular expressions here for such simple parsing grammars. Regexp is like a cancer spreading throughout the code once it was put somewhere and here it is.
All you need is to trim the string and check whether it ends with `snm/h` or `snm/s`.